pkg/services/dashboards/accesscontrol.go

```diff
@@ -173,15 +174,14 @@ func resolveDashboardScope(ctx context.Context, db Store, folderDB FolderStore,
 	return result, nil
 }
 
-func getInheritedScopes(ctx context.Context, orgID int64, folderUID string, folderSvc folder.Service) ([]string, error) {
+func GetInheritedScopes(ctx context.Context, orgID int64, folderUID string, folderSvc folder.Service) ([]string, error) {
 	ancestors, err := folderSvc.GetParents(ctx, folder.GetParentsQuery{
 		UID:   folderUID,
 		OrgID: orgID,
 	})
 
 	if err != nil {
-		// TODO return a specific error
-		return nil, err
+		return nil, fmt.Errorf("could not retrieve folder parents: %w", err)
```

Comment:
I would suggest returning an [errutil.Error](https://github.com/grafana/grafana/blob/main/contribute/backend/errors.md#errors) instead

Reply:
Good shout, I've changed the error to use `ErrInternal` from the folder package.